components/eventing-controller/pkg/backend/jetstreamv2/jetstream.go

```diff
@@ -266,29 +266,35 @@ func (js *JetStream) checkJetStreamConnection() error {
 // syncSubscriptionTypes syncs the Kyma subscription types with NATS subscriptions.
 func (js *JetStream) syncSubscriptionTypes(subscription *eventingv1alpha2.Subscription) error {
 	for key, jsSub := range js.subscriptions {
-		err := js.syncSubscriptionType(key, subscription, jsSub)
-		if err != nil {
-			return err
+		if isJsSubAssociatedWithKymaSub(key, subscription) {
+			err := js.syncSubscriptionType(key, subscription, jsSub)
+			if err != nil {
+				return err
+			}
 		}
 	}
 	return nil
 }
 
 func (js *JetStream) syncSubscriptionType(key SubscriptionSubjectIdentifier,
 	subscription *eventingv1alpha2.Subscription, subscriber Subscriber) error {
-	if !isJsSubAssociatedWithKymaSub(key, subscription) {
+	// runtime subscription is invalid and its subject exists in subscription CR filters
```

Comment:
this is just rephrasing the code. could you please describe what you actually want to check here and maybe why

Reply:
I really don't understand why for v1alpha2 implementation, validity of runtime subscribe is checked. 
As for v1alpha1 implementation, the [function](https://github.com/kyma-project/kyma/blob/718e13b75bb482cb519e34a53b1b03493b2cc915/components/eventing-controller/pkg/backend/nats/jetstream/jetstream.go#L464) could be the reason.